libs/test/SmartSyncTest/src/com/salesforce/androidsdk/smartsync/manager/ManagerTestCase.java

```diff
@@ -131,5 +143,70 @@ private RestClient initRestClient() throws Exception {
                 null, null, null, null, null, null);
         return new RestClient(clientInfo, authToken, httpAccess, null);
     }
-    
+
+    /**
+     * Helper methods to create "count" of test records
+     * @param count
+     * @return map of id to name for the created accounts
+     * @throws Exception
+     */
+    public Map<String, String> createRecordsOnServer(int count, String objectType) throws Exception {
+        Map<String, String> idToValues = new HashMap<String, String>();
+        for (int i = 0; i < count; i++) {
+
+            // Request.
+            String fieldValue = "";
+            Map<String, Object> fields = new HashMap<String, Object>();
+            //add more object type if need to support to use this API
+            //to create a new record on server
+            switch (objectType) {
+                case Constants.ACCOUNT:
+                    fieldValue = createRecordName(objectType);
+                    fields.put(Constants.NAME, fieldValue);
+                    break;
+                case Constants.OPPORTUNITY:
+                    fieldValue = createRecordName(objectType);
+                    fields.put(Constants.NAME, fieldValue);
+                    fields.put("StageName", "Prospecting");
+                    DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+                    fields.put("CloseDate", formatter.format(new Date()));
+                    break;
+                default:
+                    break;
+
+            }
+
+            RestRequest request = RestRequest.getRequestForCreate(ApiVersionStrings.getVersionNumber(targetContext), objectType, fields);
+
+            // Response.
+            RestResponse response = restClient.sendSync(request);
+            String id = response.asJSONObject().getString(LID);
+            idToValues.put(id, fieldValue);
+        }
+        return idToValues;
+    }
+
+    /**
+     * Delete records specified in idToNames
+     * @param ids
+     * @throws Exception
+     */
+    public void deleteRecordsOnServer(String[] ids, String objectType) throws Exception {
+        for (String id : ids) {
+            RestRequest request = RestRequest.getRequestForDelete(ApiVersionStrings.getVersionNumber(targetContext), objectType, id);
+            restClient.sendSync(request);
+        }
+    }
+
+    /**
+     * @return record name of the form SyncManagerTest<random number left-padded to be 8 digits long>
+     */
+    @SuppressWarnings("resource")
+    public String createRecordName(String objectType) {
+        StringBuilder sb = new StringBuilder();
+        Formatter formatter = new Formatter(sb, Locale.US);
+        formatter.format("SyncManagerTest_%s_%08d", objectType, (int) (Math.random()*10000000));
```

Comment:
Or you could just use ```System.currentTimeMillis()``` to get current millisecond timestamp, and append that instead. ```Formatter``` seems a bit heavy for this :-)